packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/opensignin.md

```diff
@@ -1,26 +1,44 @@
-Sign the user in by updating the `signIn` function of `ContentView.swift`:
+Update the `signIn` function of `ContentView.swift` with the code to sign in the user:
 
 ```swift
 func signIn() {
-  guard let rootViewController = UIApplication.shared.rootViewController else {
-    showError(title: "Unable to Sign In",
-    message: "An unkown error occured. Try restarting the app.")
-    return
-  }
+   Task {
+      do {
+         if let token = try await WebAuthentication.shared?.signIn(from: nil)
+         {
+            try Credential.store(token)
+            updateStatus("Signed in", infoText: "", signedInStatus: true)
+         }
+      } catch {
+         // Handle errors
+         showError(title: "Unable to Sign In", error: error)
+      }
+   }
+}
+```
+
+The app hands the sign-in flow to the Okta SDK. The SDK presents the sign-in web view and waits for a result. If the sign-in is succesful, the app first stores the token using `Credential.store(_)`, and then updates the user interface by setting the `signedIn` state variable to `true`.
```

Comment:
Is it worth mentioning here what the SDK displays if the sign-in is unsuccessful?

Reply:
Or linking to relevant error handling content?

Reply:
For the first part, no as that's behavior that could change and has no impact on the code written by the developer.

I did think about looking for places to link to regarding error handling, but as a QuickStart I wanted to keep things to the minimum needed.

The other things is, link to what? If you do still think linking is needed, then what Okta content could I link to? In addition, on iOS the error code is either specific to Okta (defined in the SDK, in which case it's documented somewhere in the SDK so not linkable) or a generic iOS error such as a timeout, which is in the Apple documentation, (though specific error codes may be in different frameworks.) A more general "how to handle errors on iOS" also seems like something a developer should have access to on the Apple site... and external links are always a danger as things can change.